src/lightning/data/streaming/reader.py

```diff
@@ -65,25 +72,34 @@ def _delete(self, chunk_index: int) -> None:
     def run(self) -> None:
         while True:
             with self._lock:
-                if len(self._chunks_index_to_be_processed) == 0 and len(self._chunks_index_to_be_deleted) == 0:
-                    sleep(0.005)
+                # Wait for something to do
+                if len(self._chunks_index_to_be_downloaded) == 0 and len(self._chunks_index_to_be_deleted) == 0:
+                    sleep(0.01)
                     continue
 
                 # Delete the chunks if we are missing disk space.
-                # Check every 10 items to avoid losing too much time
-                if self._max_cache_size and len(self._chunks_index_to_be_deleted) > 10:
+                if self._max_cache_size and self._processed_chunks_counter >= self._pre_download:
                     if shutil.disk_usage(self._config._cache_dir).total >= self._max_cache_size:
                         for chunk_index in self._chunks_index_to_be_deleted:
-                            if chunk_index not in self._chunks_index_to_be_processed:
+                            if chunk_index not in self._chunks_index_to_be_downloaded:
                                 self._delete(chunk_index)
+                                self._delete_chunks += 1
+                                self._processed_chunks_counter = 0
                     self._chunks_index_to_be_deleted = []
 
-                if len(self._chunks_index_to_be_processed) == 0:
+                # If there is no chunks to download, go back to waiting
+                if len(self._chunks_index_to_be_downloaded) == 0:
                     continue
```

Comment:
this continue doesn't have a sleep, is it intentional?

Reply:
Yes, because it would possibly loop back and sleep at the top.